canary/archival.go

```diff
@@ -158,6 +190,85 @@ func archivalExternalWorkflow(ctx workflow.Context, scheduledTimeNanos int64) er
 	return profile.end(nil)
 }
 
-func archivalLargeResultActivity() ([]byte, error) {
+func largeResultActivity() ([]byte, error) {
 	return make([]byte, resultSize, resultSize), nil
 }
+
+func visibilityArchivalActivity(ctx context.Context, scheduledTimeNanos int64) error {
+	scope := activity.GetMetricsScope(ctx)
+	var err error
+	scope, sw := recordActivityStart(scope, activityTypeVisibilityArchival, scheduledTimeNanos)
+	defer recordActivityEnd(scope, sw, err)
+
+	client := getActivityArchivalContext(ctx).cadence
+	resp, err := client.Describe(ctx, archivalDomain)
+	if err != nil {
+		return err
+	}
+
+	if resp.Configuration != nil &&
```

Comment:
Why would this occur?

Reply:
I don't think this will occur. Including this check for safety so that the code won't panic. I can remove if you have strong opinion on this.

Same below.